fortanix-vme/eif-tools/src/bin/ftxvme-runner-cargo.rs

```diff
@@ -0,0 +1,302 @@
+            .package
+            .map(|package| {
+                package.metadata
+                    .map(|metadata| metadata.fortanix_vme)
+            })
+        .flatten();
+
+        let config = if let Some(fortanix_vme_metadata) = fortanix_vme_metadata {
+
+            // Use default cpu count if metadata doesn't specify cpu_ids or cpu_count.
+            // Also, specifying both cpu_count and cpu_ids is invalid but we don't handle it here
+            // because nitro-cli will anyway error out if that's the case.
+            let (cpu_ids, cpu_count) = match (fortanix_vme_metadata.cpu_ids, fortanix_vme_metadata.cpu_count) {
+                (None, None) => (None, Some(FortanixVmeConfig::DEFAULT_CPU_COUNT)),
+                val => val,
+            };
+
+            FortanixVmeConfig {
+                cpu_count,
+                cpu_ids,
+                .. fortanix_vme_metadata
+            }
+        } else {
+            Default::default()
+        };
+
+        Ok(config)
+    }
+
+}
+
+impl Default for FortanixVmeConfig {
+    fn default() -> Self {
+        Self {
+            cpu_count: Some(FortanixVmeConfig::DEFAULT_CPU_COUNT),
+            memory: FortanixVmeConfig::DEFAULT_MEMORY,
+            debug_mode: FortanixVmeConfig::DEFAULT_DEBUG_MODE,
+            enclave_name: None,
+            cpu_ids: None,
+            enclave_cid: None,
+            nitro_cli_run_enclave_config: None,
+            verbose: false,
+            eif_file_path: FortanixVmeConfig::default_eif_path(),
+            resource_path: None,
+            signing_certificate: None,
+            private_key: None,
+            disable_fortanix_vme_runner_start: false,
+        }
+    }
+}
+
+fn main() -> anyhow::Result<()> {
+    let fortanix_vme_config = FortanixVmeConfig::get()?;
+
+    let ftxvme_elf2eif = command! {
+        "ftxvme-elf2eif" => args(
+            "--input-file"     => &ARGS[1],
+            "--output-file"    => &fortanix_vme_config.eif_file_path,
+            "--verbose"        => ?is_true(fortanix_vme_config.verbose),
+            "--resource-path"  => ?is_some(fortanix_vme_config.resource_path)
+        )
+    };
+
+    run_command(ftxvme_elf2eif)?;
+
+    if !fortanix_vme_config.disable_fortanix_vme_runner_start {
+        // We just try to start fortanix-vme-runner and don't wait on it. So,
+        // we don't know if it errors out. I think this should be enough
+        // as purpose of the runner is to provide easy setup. If someone really wants to
+        // be sure, they can set `disable_fortanix_vme_runner_start` to true and start
+        // it manually.
+        //
+        // If required, other option can be to link fortanix-vme-runner as a library and spawn the
+        // proxy server in another thread. The thing here will be that if fortanix-vme-runner
+        // already running, it will error out will addr in use. To be sure, we could probably
+        // communicate with fortanix-vme-runner.
```

Comment:
Why would we want to disable the runner from starting? Yes starting multiple runners will currently error out because it won't be able to bind to vsock port 10000, but that'll change in the future; we'll ensure that the runner contacts the enclave, instead of the other way around. I'd remove this option to avoid having to support it in the future.

Reply:
Makes sense, I will remove this option.